### DISCUSSION TITLE
Initial Connection/Query to Zilliz Instance Takes ~20s, Then is Fast

### DISCUSSION THREAD

**Discussion by tieandrews** [milvus-io/milvus#31916] [Q&A and General discussion]
I'm not sure this is the right spot for Zilliz questions but I was pointed here by Zilliz so here goes.

I have a python web app that creates a connection to the Zilliz instance, then the very first query against it consistently takes 15-20 seconds, then after that it's a usable 200-400ms per query.

My cluster is on GCP in US-West-1 and my web app is Azure App Service can-central-1, and I know this will make the connection slower and would be totally ok for a 4-8s initial connection/query, but 20s seems a bit silly.

I've looked at creating the private network connections in Zilliz and what it involves is just too much cost/effort right now and I'm hoping there's something I can do to improve this connection time.

Thanks in advance for any help!

**Comment by yhmo** [comment]
Sounds like a lazy-load behavior, wait Zilliz engineers to confirm.

**Comment by xiaofan-luan** [comment]
> I'm not sure this is the right spot for Zilliz questions but I was pointed here by Zilliz so here goes.
> 
> I have a python web app that creates a connection to the Zilliz instance, then the very first query against it consistently takes 15-20 seconds, then after that it's a usable 200-400ms per query.
> 
> My cluster is on GCP in US-West-1 and my web app is Azure App Service can-central-1, and I know this will make the connection slower and would be totally ok for a 4-8s initial connection/query, but 20s seems a bit silly.
> 
> I've looked at creating the private network connections in Zilliz and what it involves is just too much cost/effort right now and I'm hoping there's something I can do to improve this connection time.
> 
> Thanks in advance for any help!

Could you ping your endpoint and see how long does it take to loop back? 

Also if you could offer test code that could even help. 

For Zilliz cloud users you can always submit a ticket https://support.zilliz.com/hc/en-us/requests/new and you will get more instant response compred to community support
